src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java

```diff
@@ -127,18 +127,30 @@ public void runLambdaTest() {
     // setter methods for the properties
 
     public void setUsername(String username) {
+        if (username == null || username.trim().isEmpty()) {
+            throw new IllegalArgumentException("Username cannot be null or empty");
+        }
         this.username = username;
     }
 
     public void setAccessKey(String accessKey) {
+        if (accessKey == null || accessKey.trim().isEmpty()) {
+            throw new IllegalArgumentException("Access key cannot be null or empty");
+        }
         this.accessKey = accessKey;
     }
 
     public void setAppFilePath(String appFilePath) {
+        if (appFilePath == null || appFilePath.trim().isEmpty()) {
```

Comment:
```suggestion
        if (appFilePath != null && appFilePath.trim().isEmpty()) {
	        throw new IllegalArgumentException("App file path cannot be empty");
	    }
```

`appFilePath` and `testSuiteFilePath` are not strictly necessary and are optional and adding these checks will break functionality.
Users can also provide pre-uploaded IDs instead of file paths. And, either `appFilePath` or `testSuiteFilePath` could be provided. With this code it will fail even when one of the path is provided.

Same for the other file. Do check other remaining possibilities/combinations.

Reply:
I have updated the code